src/MsgPack/Serialization/SerializationCompatibilityOptions.cs

```diff
@@ -126,6 +126,43 @@ public bool IgnorePackabilityForCollection
 			}
 		}
 
+#if NETFX_35 || UNITY || SILVERLIGHT
+		private volatile bool _alwaysAssumeCollections;
+#else
+		private bool _alwaysAssumeCollections;
+#endif // NETFX_35 || UNITY || SILVERLIGHT
+
+		/// <summary>
+		///		Gets or sets a value indicating whether serializer generator should always try to serialize a type implementing IEnumerable as a collection or not.
+		/// </summary>
+		/// <value>
+		///		<c>true</c> if serializer generator should serialize a type implementing IEnumerable as only a collection; otherwise, <c>false</c>. The default is <c>false</c>.
+		/// </value>
+		/// <remarks>
+		///		Historically, MessagePack for CLI always tried to serialize any type that implemented IEnumerable as a collection, throwing an exception
+		///		if an Add method could not be found. However, for types that implement IEnumerable but don't have an Add method the generator will now
+		///		serialize the type as a non-collection type. To restore the old behavior for backwards compatibility, set this option to <c>false</c>.
+		/// </remarks>
+		public bool AlwaysAssumeCollections
```

Comment:
I think the meaning of this property is opposite.
If the serializer generator "always assume collections", thus it also assume "collection should implement Add".
This name should be `AllowEnumerableObject`(alternative name is welcome).